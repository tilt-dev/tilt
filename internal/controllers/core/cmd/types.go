package cmd

import "github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"

type Cmd = v1alpha1.Cmd
type CmdSpec = v1alpha1.CmdSpec
type CmdStatus = v1alpha1.CmdStatus
type CmdStateRunning = v1alpha1.CmdStateRunning
type CmdStateTerminated = v1alpha1.CmdStateTerminated

const LabelManifest = v1alpha1.LabelManifest
