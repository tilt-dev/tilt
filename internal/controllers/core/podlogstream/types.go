package podlogstream

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PodLogStream = v1alpha1.PodLogStream
type PodLogStreamSpec = v1alpha1.PodLogStreamSpec
type PodLogStreamStatus = v1alpha1.PodLogStreamStatus
type ContainerLogStreamStatus = v1alpha1.ContainerLogStreamStatus
type ObjectMeta = metav1.ObjectMeta
