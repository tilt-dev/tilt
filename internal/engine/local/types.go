package local

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type Cmd = v1alpha1.Cmd
type CmdList = v1alpha1.CmdList
type CmdStatus = v1alpha1.CmdStatus
type CmdSpec = v1alpha1.CmdSpec
type CmdStateWaiting = v1alpha1.CmdStateWaiting
type CmdStateTerminated = v1alpha1.CmdStateTerminated
type CmdStateRunning = v1alpha1.CmdStateRunning
type ObjectMeta = metav1.ObjectMeta
