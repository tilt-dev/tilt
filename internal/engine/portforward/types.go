package portforward

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PortForward = v1alpha1.PortForward
type PortForwardSpec = v1alpha1.PortForwardSpec
type PortForwardStatus = v1alpha1.PortForwardStatus
type ObjectMeta = metav1.ObjectMeta
type Forward = v1alpha1.Forward
