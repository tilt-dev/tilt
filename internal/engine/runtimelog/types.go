package runtimelog

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PodLogStream = v1alpha1.PodLogStream
type PodLogStreamSpec = v1alpha1.PodLogStreamSpec
type ObjectMeta = metav1.ObjectMeta
