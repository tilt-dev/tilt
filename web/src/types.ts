import { Alert } from "./alerts"
import { Facet } from "./facets"

export enum SocketState {
  Loading,
  Reconnecting,
  Closed,
  Active,
}

export enum ResourceView {
  Log,
  Alerts,
  Facets = 2,
}

export enum TriggerMode {
  TriggerModeAuto,
  TriggerModeManualAfterInitial,
  TriggerModeManualIncludingInitial,
}

// what is the status of the resource in the cluster
export enum RuntimeStatus {
  Ok = "ok",
  Pending = "pending",
  Error = "error",
  NotApplicable = "not_applicable",
}

// What is the status of the resource with respect to Tilt
export enum ResourceStatus {
  Building, // Tilt is actively doing something (e.g., docker build or kubectl apply)
  Pending, // not building, healthy, or unhealthy, but presumably on its way to one of those (e.g., queued to build, or ContainerCreating)
  Healthy, // e.g., build succeeded and pod is running and healthy
  Unhealthy, // e.g., last build failed, or CrashLoopBackOff
  Warning, // e.g., an undismissed restart
  None, // e.g., a manual build that has never executed
}

export type SnapshotHighlight = {
  beginningLogID: string
  endingLogID: string
  text: string
}

export enum ShowFatalErrorModal {
  Default,
  Show,
  Hide,
}

export type Snapshot = {
  view: Proto.webviewView
  isSidebarClosed: boolean
  path?: string
  snapshotHighlight?: SnapshotHighlight | null
}

// A plaintext representation of a line of the log,
// with metadata to render it in isolation.
//
// The metadata should be stored as primitive fields
// so that React's default caching behavior will kick in.
export type LogLine = {
  // We assume that 'text' does not contain a newline
  text: string
  manifestName: string
  level: string
}
