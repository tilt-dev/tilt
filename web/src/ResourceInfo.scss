@import "constants.scss";

.resourceInfo {
  display: flex;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
}
.resourceInfo + .resourceInfo {
  margin-left: $spacing-unit / 2;
  border-left: 1px dotted $color-gray-light;
  padding-left: $spacing-unit / 2;
}
.resourceInfo-label {
  text-transform: uppercase;
  color: $color-gray-light;
  font-weight: bold;
  margin-right: $spacing-unit / 4;
}
.resourceInfo-value {
  display: inline-block;
  font-family: $font-monospace;
  user-select: all;
  margin: 0;
  text-overflow: ellipsis;
  overflow: hidden;
}
a.resourceInfo-value {
  text-decoration: none;
}
a.resourceInfo-value:hover {
  text-decoration: underline;
}
.resourceInfo-value + .resourceInfo-value::before {
  content: "";
  margin-left: $spacing-unit / 2;
  padding-left: $spacing-unit / 2;
  height: $spacing-unit / 2;
  display: inline-block;
  border-left: 1px solid $color-gray-light;
}
