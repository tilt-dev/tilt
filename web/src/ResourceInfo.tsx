import React, { PureComponent } from "react"
import styled from "styled-components"
import * as s from "./style-helpers"
import { SnapshotHighlight } from "./types"
import { ReactComponent as SnapshotSvg } from "./assets/svg/snapshot.svg"

type HUDHeaderProps = {
  podID?: string
  endpoints?: string[]
  podStatus?: string
  showSnapshotButton: boolean
  highlight: SnapshotHighlight | null
  handleOpenModal: () => void
}

let Root = styled.div`
  display: flex;
  align-items: center;
  height: ${s.Height.HUDheader - s.Height.secondaryNav}px;
  padding-left: ${s.SizeUnit(0.5)};
  padding-right: ${s.SizeUnit(0.25)};
`
let ResourceInfoStyle = styled.div`
  flex: 1;
  display: flex;
  padding-right: ${s.SizeUnit(0.5)};
`

let PodStatus = styled.span`
  font-family: ${s.Font.sansSerif};
  font-size: ${s.FontSize.small};
  flex: 1;
`

let PodId = styled.span``

let PortForward = styled.span`
  ${PodId} + & {
    margin-left: ${s.SizeUnit(0.5)};
    border-left: 1px solid ${s.Color.gray};
    padding-left: ${s.SizeUnit(0.5)};
  }
`

let PortForwardLabel = styled.span`
  color: ${s.Color.grayLight};
  margin-right: ${s.SizeUnit(0.25)};

  ${s.mixinHideOnSmallScreen}
`

let PortForwardLink = styled.a`
  & + & {
    padding-left: ${s.SizeUnit(0.25)};
    border-left: 1px dotted ${s.Color.grayLight};
    margin-left: ${s.SizeUnit(0.25)};
  }
`

let SnapshotButton = styled.button`
  border: 1px solid transparent;
  font-family: ${s.Font.sansSerif};
  font-size: ${s.FontSize.smallest};
  background-color: transparent;
  color: ${s.Color.blue};
  display: flex;
  align-items: center;
  box-sizing: border-box;
  padding-left: ${s.SizeUnit(0.5)};
  padding-right: ${s.SizeUnit(0.5)};
  padding-top: ${s.SizeUnit(0.25)};
  padding-bottom: ${s.SizeUnit(0.25)};
  transition: border-color;
  transition-duration: ${s.AnimDuration.default};
  text-decoration: none;
  cursor: pointer;

  &:hover {
    background-color: ${s.Color.grayDark};
    border-color: ${s.Color.blue};
  }

  &.isHighlighted {
    border-color: ${s.Color.blue};
  }
`

let SnapshotButtonSvg = styled(SnapshotSvg)`
  margin-right: ${s.SizeUnit(0.25)};
`

class ResourceInfo extends PureComponent<HUDHeaderProps> {
  renderSnapshotButton() {
    let highlight = this.props.highlight

    if (this.props.showSnapshotButton)
      return (
        <SnapshotButton
          onClick={this.props.handleOpenModal}
          className={`snapshotButton ${highlight ? "isHighlighted" : ""}`}
        >
          <SnapshotButtonSvg />
          <span>
            Create a <br />
            Snapshot
          </span>
        </SnapshotButton>
      )
  }

  render() {
    let podStatus = this.props.podStatus
    let podID = this.props.podID

    let endpoints = this.props.endpoints ?? []
    let endpointsEl = endpoints?.length > 0 && (
      <PortForward>
        <PortForwardLabel>
          Port-Forward{endpoints?.length > 1 ? "s" : ""}:
        </PortForwardLabel>
        {endpoints?.map(ep => (
          <PortForwardLink
            href={ep}
            target="_blank"
            rel="noopener noreferrer"
            key={ep}
          >
            {ep}
          </PortForwardLink>
        ))}
      </PortForward>
    )

    return (
      <Root>
        <ResourceInfoStyle>
          <PodStatus>{podStatus}</PodStatus>
          {podID && <PodId>{podID}</PodId>}
          {endpointsEl}
        </ResourceInfoStyle>
        {this.renderSnapshotButton()}
      </Root>
    )
  }
}

export default ResourceInfo
