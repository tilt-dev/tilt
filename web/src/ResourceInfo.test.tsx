import React from "react"
import renderer from "react-test-renderer"
import { MemoryRouter } from "react-router"
import { ResourceView, SnapshotHighlight } from "./types"
import ResourceInfo from "./ResourceInfo"

const fakeHandleOpenModal = () => {}

it("shows snapshot url", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ResourceInfo
          showSnapshotButton={true}
          handleOpenModal={fakeHandleOpenModal}
          highlight={null}
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it("doesn't render snapshot button if it's a snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ResourceInfo
          showSnapshotButton={false}
          handleOpenModal={fakeHandleOpenModal}
          highlight={null}
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
