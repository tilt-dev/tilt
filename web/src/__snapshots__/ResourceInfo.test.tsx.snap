// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't render snapshot button if it's a snapshot 1`] = `
<div
  className="sc-bdVaJa kyeOVJ"
>
  <div
    className="sc-bwzfXH fdpnBr"
  >
    <span
      className="sc-htpNat dgyfiu"
    />
  </div>
</div>
`;

exports[`shows snapshot url 1`] = `
<div
  className="sc-bdVaJa kyeOVJ"
>
  <div
    className="sc-bwzfXH fdpnBr"
  >
    <span
      className="sc-htpNat dgyfiu"
    />
  </div>
  <button
    className="sc-gzVnrw jlKHQi snapshotButton "
    onClick={[Function]}
  >
    <svg
      className="sc-htoDjs jYtsma"
    >
      snapshot.svg
    </svg>
    <span>
      Create a 
      <br />
      Snapshot
    </span>
  </button>
</div>
`;
